### DISCUSSION TITLE
[Bug]: Playback UI breaks in maximized windows in larger screen resolutions

### DISCUSSION THREAD

**Discussion by DReffects** [blakeblackshear/frigate#12223] [Report a Bug]
### Describe the problem you are having

When using an UHD Screen (3840x2160), in my case with 125% scaling in Windows, since it's a 43 inch screen, the playback screen ui breaks.

there are two distinct issues:

- Double-Clicking the playback or live video breaks the player.
- Playback UI Elements are missing when window is maximized

### Steps to reproduce

Zoom issue:
If you double click the playback window multiple times, the screen turns black and a red error message is being shown briefly at the top (too short to capture). Refreshing the page restores to normal. This is no issue when having frigate in a smaller browser window. It only happens when maximized. 

UI Elements missing:
If I maximize the browser window during playback of recordings the pause, play, speed controls + the other cams are missing. 
![image](https://github.com/blakeblackshear/frigate/assets/48361672/73613832-3315-4ba1-b21a-3f4da4e30d23)

### Version

0.14.0-4e5a6eb

### Frigate config file

```yaml
-
```

### Relevant log output

```shell
-
```

### Operating system

HassOS

### Install method

HassOS Addon

### Network connection

Wired

### Camera make and model

Dahua

### Any other information that may be helpful

Chrome Version 126.0.6478.127 (Official Build) (64-Bit)

**Comment by hawkeye217** [comment]
Thanks for the bug report. I can see a potential problem with the code. If you return scaling to 100% temporarily, are the elements visible?

**Reply by DReffects** [reply]
Sure thing!
No, returning scaling to 100% does not help. Also found out, that it starts to happen when the window is wide enough. If I make it about 90% of the width of the screen, the elements disappear. 
Also the camera image is not being scaled properly and is cropped at top and bottom. 

I've attached a video showing the issue. 

https://github.com/blakeblackshear/frigate/assets/48361672/6e2f7cf9-c9cb-4098-a980-ce0003679775

:-)

**Reply by hawkeye217** [reply]
I changed my display to 3840x2160 and used Chrome 126 on the beta 3 code and have been unable to reproduce this when looking at the history recordings of my 16x9 cameras.

What's the resolution of the camera in question? Does it happen on all of your cameras?

**Reply by DReffects** [reply]
Thanks for checking. My Cameras have 3072x2048 resolution. All have the same resolution and are the same model.

**Reply by hawkeye217** [reply]
Thanks. The aspect ratio of your cameras is 1.5, which would explain why there's an issue.

**Reply by DReffects** [reply]
Ah, I see. That's the most common aspect for Dahua cameras.

**Reply by NickM-27** [reply]
I don't believe that is true, most are 16:9 AKA 1.7

**Reply by DReffects** [reply]
Depends very much on the camera line. Most "home user" cameras are indeed 16:9, but with many of the more professional cameras with 5MP or more you can utilize the full sensor which is usually in 1.5 or 1.33 ratio.

https://www.megateh.eu/products/3072%202048? - 52 Camera models in 1.5 ratio
https://www.megateh.eu/products/2592%201944? - 79 camera models in 1.33 ratio
(Those are only the 5 and 6MP models alone). 

most 12MP 360° fishey cameras use 4000x3000 

substreams with dahua usually are at 704x576 / 1.22 when you utilize smart codec.

I think there's really no standard with the resolution of the cameras.

**Reply by hawkeye217** [reply]
I made a PR with some fixes that could potentially solve your issue: https://github.com/blakeblackshear/frigate/pull/12297

The fixes are available in the latest dev build and will also be available in beta 4.

**Reply by DReffects** [reply]
Sweet, Thanks!! I'll give it a try when beta 4 becomes available. Or is there an easy way to try/test the dev-build? :-)

**Reply by hawkeye217** [reply]
If you're running as a HASS add-on, then no - you'll have to wait until beta 4. If not, you can pull a dev build for Docker from here: https://github.com/blakeblackshear/frigate/pkgs/container/frigate

**Reply by hawkeye217** [reply]
Related question - what is the resolution of the substream on the problematic camera?
